lib/iris/coords.py

```diff
@@ -1373,33 +1446,26 @@ def from_regular(cls, zeroth, step, count, standard_name=None,
             bounds values will be defined. Defaults to False.
 
         """
-        coord = DimCoord.__new__(cls)
-
-        coord.standard_name = standard_name
-        coord.long_name = long_name
-        coord.var_name = var_name
-        coord.units = units
-        coord.attributes = attributes
-        coord.coord_system = coord_system
-        coord.circular = circular
-
-        points = (zeroth+step) + step*np.arange(count, dtype=np.float32)
+        points = (zeroth + step) + step * np.arange(count, dtype=np.float32)
+        # Spoof `iris.util.is_regular`, which only works on coords (which we
+        # don't yet have).
+        diffs = np.diff(points)
+        if not np.allclose(diffs, np.mean(diffs), rtol=0.001):
+            points = (zeroth + step) + step * np.arange(count,
+                                                        dtype=np.float64)
```

Comment:
By the way, don't let the fact that I'm continuing to discuss this let you think I'm not happy with the method as it now is. Unless anyone else has concerns I'm, happy for it to go in as-is.